pkg/api/query/v1.go

```diff
@@ -224,6 +227,19 @@ func (qapi *QueryAPI) parsePartialResponseParam(r *http.Request, defaultEnablePa
 	return defaultEnablePartialResponse, nil
 }
 
+func (qapi *QueryAPI) parseStep(r *http.Request, defaultRangeQueryStep time.Duration) (step time.Duration, _ *api.ApiError) {
```

Comment:
I think this does not fix the issue (but is a very good try!) (: 

The reason is that there is no default set by server. As you could see before your PR server was erroring out when step is not provided. I think that's ok to keep this is way.

To change default I think we need to alter frontend code - as this is where the default 1s is injected. That would be amazing to do! WDYT? (:

Reply:
Yes I agree 😊 . This flag is always going to be overwritten. So we need to provide some kind of input field on front-end using which user can set default step ? If yes, Can i work on that as well ?

Reply:
Of course, feel free to use the same PR